src/Agent/NewRelic/Agent/Core/Wrapper/WrapperMap.cs

```diff
@@ -132,5 +162,227 @@ private static bool CanWrap(InstrumentedMethodInfo instrumentedMethodInfo, IWrap
 
             return canWrapResponse.CanWrap;
         }
+
+        private delegate IMethodWrapper WrapperFactory(IAgent agentWrapperApi, ITransaction transaction, InstrumentedMethodCall instrumentedMethodCall, object invocationTarget);
+
+        public static IWrapper[] CreateMethodWrappers(Type wrapperType)
```

Comment:
This code runs at agent start up and locates methods decorated with `MethodWrapper` attributes. It provides a different way of mapping instrumented methods to wrappers.